truenas/webui
# NAS-110126 / 21.08 / Selecting multiple directories for source path for push and pull

We need to allow selecting multiple source directories for cloud sync (local directories for PUSH, remote directories for PULL). All source directories must have the same parent (be on the same level).

When only one directory is selected, the behavior is the same. However, when multiple directories are selected:

1. `path` (for PUSH) or `attributes.folder` (for PULL) is set to the parent directory path.

2. New `include` field is populated with list of the names of the selected directories in the following form: ["/Directory 1/**", "/Directory 2/**"].

For example in the following tree:
![image](https://user-images.githubusercontent.com/24204846/124230973-240e6f80-db29-11eb-8d2c-6d2d4a25fe7a.png)

We should be able to select `sub1.1` and `sub1.2` (but not `sub1.1` and `sub2`). If we select them, we'll get:
`
{
    "path": "/mn/pool/sub1",
    "include": ["/sub1.1/**", "/sub1.2/**"]
}
`


To test,

Select multiple directories when creating a cloud sync task. Submit the form. The payload should be according to the description.

NAS-110126
